.tools/validate.go

```diff
@@ -70,7 +76,7 @@ func main() {
 			fmt.Println("PASS")
 			if *flVerbose {
 				for _, r := range vr {
-					if !r.Pass {
+					if r.Pass {
```

Comment:
On Thu, Sep 10, 2015 at 06:25:33AM -0700, Vincent Batts wrote:
> > -					if !r.Pass {
> > +					if r.Pass {
> 
> not true. this is so that you can see the messages for passed tests
> as well.

You have (from #167):

    if _, fail := vr.PassFail(); fail == 0 {
      fmt.Println("PASS")
      if *flVerbose {
        for _, r := range vr {
          if !r.Pass {
            fmt.Printf("  - %s\n", r.Msg)
          }
        }
      }
    } else {
      fmt.Println("FAIL")
      // default, only print out failed validations
      for _, r := range vr {
        if !r.Pass {
          fmt.Printf("  - %s\n", r.Msg)
        }
      }
    }

The fix you're proposing here would be the same as:

    if _, fail := vr.PassFail(); fail == 0 {
      fmt.Println("PASS")
      if *flVerbose {
        for _, r := range vr {
          fmt.Printf("  - %s\n", r.Msg)
        }
      }
    } else {
      fmt.Println("FAIL")
      // default, only print out failed validations
      for _, r := range vr {
        if !r.Pass {
          fmt.Printf("  - %s\n", r.Msg)
        }
      }
    }

So keep the Printf, but drop the r.Pass check from the ‘fail== 0’
branch (because we know that none of the checks failed).

If you want to see passes in all cases, I think you want:

    if _, fail := vr.PassFail(); fail == 0 {
      fmt.Println("PASS")
    } else {
      fmt.Println("FAIL")
    }
    if *flVerbose {
      for _, r := range vr {
        if r.Pass {
          fmt.Printf("ok")
        }
        if r.Pass {
          fmt.Printf("not ok")
        }
        fmt.Println(" %s", r.Msg)
      }
    }